csharp/Adapter/Microsoft.Spark.CSharp/Proxy/Ipc/SparkContextIpcProxy.cs

```diff
@@ -361,6 +361,18 @@ public IColumnProxy CreateWindowFunction(string name)
             }
             return jmap;
         }
+
+        public static JvmObjectReference GetJavaHashMap<K, V>(IEnumerable<KeyValuePair<K, V>> enumerable)
```

Comment:
I agree that a `Dictionary` (or actually `Dictionary<sting,string>`) seems exactly what is needed for the `conf ` argument. However the same is true for the previously used `GetJavaMap` (which created a `java.util.Hashtable`), and maybe this optimization you suggest should be done in an additional/different PR?
Thanks!